utam-compiler/src/main/java/utam/compiler/helpers/TypeUtilities.java

```diff
@@ -122,16 +118,39 @@ public static Element asBasicType(String jsonString) {
     }
 
     public static boolean isBasicType(TypeProvider type) {
-      for (TypeUtilities.Element basicType : TypeUtilities.Element.values()) {
+      if (type instanceof Element) {
+        return true;
+      }
+      for (TypeUtilities.BasicElementInterface basicType : TypeUtilities.BasicElementInterface.values()) {
         if (basicType.isSameType(type)) {
           return true;
         }
       }
       return false;
     }
 
-    public static TypeUtilities.Element getBasicElementType(TypeProvider type) {
-      for (TypeUtilities.Element basicType : TypeUtilities.Element.values()) {
+    public static BasicElementInterface[] getBasicElementTypes(TypeProvider type) {
+      if (type instanceof Element) {
+        Element element = (Element)type;
+        BasicElementInterface[] interfaceArray = new BasicElementInterface[element.basicInterfaces.size()];
+        element.basicInterfaces.toArray(interfaceArray);
+        return interfaceArray;
```

Comment:
i think `return element.basicInterfaces.toArray(BasicElementInterface[]::new);` offers better code readability
i might be wrong though

Reply:
Today, I learned... :) Yes, this is much more readable. Fixed in next commit.